File: TeeNew/Builder2007/Surface_XYZFloat.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <math.h>
#pragma hdrstop

#include "Surface_XYZFloat.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TSurfaceFloatOther *SurfaceFloatOther;
//---------------------------------------------------------------------------
__fastcall TSurfaceFloatOther::TSurfaceFloatOther(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TSurfaceFloatOther::FormCreate(TObject *Sender)
{
  Series1->IrregularGrid = true; // <-- IMPORTANT ! means X and Z are float
  for (int x=1;x<10;x++)
    for (int z=1;z<10;z++) Series1->AddXYZ( x/10.0, sqrt(x*z), z/5.0, "", clTeeColor);

  // set palette colors to "10, strong" ...
  Series1->UseColorRange = false;
  Series1->UsePalette = true;
  Series1->PaletteStyle = psStrong;
  Series1->PaletteSteps = 10;

  // adjust some axes properties...
  Chart1->DepthAxis->Visible = true;
  Chart1->DepthAxis->AxisValuesFormat ="0.#";
  Chart1->DepthAxis->Increment = 0.2;

  Chart1->BottomAxis->AxisValuesFormat = "0.#";
  Chart1->BottomAxis->Increment = 0.1;

  // visual properties...
  Chart1->Chart3DPercent =100;
  TView3DOptions *opt = Chart1->View3DOptions;
  opt->Orthogonal = false;
  opt->Perspective = 50;
  opt->Rotation = 327;
  opt->Elevation = 352;
  opt->Zoom =70;
}
//---------------------------------------------------------------------------

File: TeeNew/Surface_XYZFloat.pas
unit Surface_XYZFloat;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, TeeSurfa, TeeProcs, Chart, TeeTools, TeeGDIPlus;

type
  TSurfaceFloatOther = class(TBaseForm)
    Series1: TSurfaceSeries;
    ChartTool1: TRotateTool;
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TSurfaceFloatOther.FormCreate(Sender: TObject);
var x,z : Integer;
begin
  inherited;

  Chart1.View3D := True;
  Series1.IrregularGrid:=True; { <-- IMPORTANT ! means X and Z are float }

  for x:=1 to 10 do
    for z:=1 to 10 do
      Series1.AddXYZ( x / 10.0, sqrt(x*z), z / 5.0);

  { set palette colors to "10, strong" ... }
  Series1.UseColorRange:=False;
  Series1.UsePalette:=True;
  Series1.PaletteStyle:=psStrong;
  Series1.PaletteSteps:=10;

  { adjust some axes properties... }
  Chart1.DepthAxis.Visible:=True;
  Chart1.DepthAxis.AxisValuesFormat:='0.#';
  Chart1.DepthAxis.Increment:=0.2;

  Chart1.BottomAxis.AxisValuesFormat:='0.#';
  Chart1.BottomAxis.Increment:=0.1;

  { visual properties... }
  Chart1.Chart3DPercent:=100;
  with Chart1.View3DOptions do
  begin
    Orthogonal:=False;
    Perspective:=50;
    Rotation:=327;
    Elevation:=352;
    Zoom:=70;
  end;
end;

initialization
  RegisterClass(TSurfaceFloatOther);
end.

